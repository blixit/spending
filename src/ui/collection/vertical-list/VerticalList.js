import React from 'react';

import BasicList from 'ui/collection/list/BasicList';

const VerticalList = ({ items, renderItem, ...rest }) => {
  return (
    <div {...rest} >
      <BasicList items={items} >
        {(item, key) => renderItem(item, key)}
      </BasicList>
    </div>
  );
};

export default VerticalList;
