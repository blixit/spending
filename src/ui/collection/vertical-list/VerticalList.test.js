import React from 'react';

import render from 'testing/render';

import VerticalList from './VerticalList';

describe('[collection/vertical-list] VerticalList', () => {
  it('renders properly', () => {
    const props = {
      items: [{
        id: 1,
        label: 'label'
      }],
      renderItem: (item, key) => (
        <div key={key}>Item content</div>
      )
    };
    const tree = render(<VerticalList {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
