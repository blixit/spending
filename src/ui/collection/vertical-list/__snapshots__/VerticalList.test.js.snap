// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[collection/vertical-list] VerticalList renders properly 1`] = `
<div>
  <div>
    Item content
  </div>
</div>
`;
